### PR TITLE
refactor: streamline & update rust dependencies

### PR DESCRIPTION
**Pull request by chadoh**
Partially satisfies https://github.com/near/devx/issues/157

Following the advice given [here] to ensure there are no different versions of common libraries

Formatting changes come from the auto run of [rustfmt] in my editor

  [here]: https://github.com/near-examples/rust-status-message/pull/13#discussion_r416946300
  [rustfmt]: https://github.com/rust-lang/rust.vim#formatting-with-rustfmt

### CODE REVIEW

**Comment by chadoh** [comment]
@evgenykuzyakov tests currently failing with:

```
test tests::test_fail_request_transfer_already_used ... FAILED
test tests::test_request_transfer_ok ... FAILED

failures:

---- tests::test_fail_request_transfer_already_used stdout ----
thread 'tests::test_fail_request_transfer_already_used' panicked at 'The given hash is already used for transfer', src/lib.rs:98:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- tests::test_request_transfer_ok stdout ----
Salt is 58
thread 'tests::test_request_transfer_ok' panicked at 'The given hash is already used for transfer', src/lib.rs:98:9
```

Pointers on how to fix?

File: contract-rs/src/lib.rs [hunk 1]
```diff
@@ -83,10 +90,16 @@ impl TransferFaucet {
...
-        assert!(num_leading_zeros(&hash) >= self.min_difficulty, "The proof is work is too weak");
+        assert!(
+            num_leading_zeros(&hash) >= self.min_difficulty,
+            "The proof is work is too weak"
+        );
 
         // Checking that the given hash is not used yet and remembering it.
-        assert!(!self.existing_hashes.insert(&hash), "The given hash is already used for transfer");
+        assert!(
+            !self.existing_hashes.insert(&hash),
```
Comment (evgenykuzyakov) [inline contract-rs/src/lib.rs]:
```suggestion
            self.existing_hashes.insert(&hash),
```
`near-sdk` at some version has a fix makes `insert` returns `true` as described in the comment

**Review by evgenykuzyakov — COMMENTED** [review]
Also replace `StrPublicKey` with  `near_sdk::json_types::Base58PublicKey`

File: contract-rs/src/lib.rs (hunk 1 above)
Comment (chadoh) [reply contract-rs/src/lib.rs]: Great, thanks!

**Comment by chadoh** [comment]
> Also replace `StrPublicKey` with  `near_sdk::json_types::Base58PublicKey`

I was able to get rid of the whole `types.rs` file after doing this 🎉

**Pull request merged**